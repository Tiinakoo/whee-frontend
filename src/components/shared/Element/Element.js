import React from "react";
import styles from "./Element.module.scss";
import classNames from "classnames";
import upperFirst from "lodash/fp/upperFirst";

const Element = ({ minWidth, padding, className, children, ...props }) => {
  const classes = classNames([
    styles.element,
    styles[`element__minWidth${upperFirst(minWidth)}`],
    styles[`element__padding${upperFirst(padding)}`],
    className,
  ]);

  return (
    <div className={classes} {...props}>
      {children}
    </div>
  );
};

export default Element;
