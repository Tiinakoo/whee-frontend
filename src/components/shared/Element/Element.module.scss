@import "../../../styles/mixins";
@import "../../../styles/baseline";

.element {
  display: block;

  &__minWidthSm {
    min-width: 8 * $baseline;

    @include md {
      min-width: 16 * $baseline;
    }
  }

  &__minWidthMd {
    min-width: 16 * $baseline;

    @include md {
      min-width: 32 * $baseline;
    }
  }

  &__minWidthLg {
    min-width: 32 * $baseline;

    @include md {
      min-width: 64 * $baseline;
    }
  }

  &__paddingSm {
    padding: 2 * $baseline 0;

    @include md {
      padding: 2 * $baseline;
    }
  }

  &__paddingMd {
    padding: 6 * $baseline 0;

    @include md {
      padding: 6 * $baseline;
    }
  }

  &__paddingLg {
    padding: 8 * $baseline 0;

    @include md {
      padding: 8 * $baseline;
    }
  }
}
