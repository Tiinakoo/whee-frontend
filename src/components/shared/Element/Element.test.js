import React from "react";

import { mount } from "enzyme";
import Element from "./Element";

describe("Element", () => {
  it("renders", () => {
    const component = mount(<Element />);

    expect(component).toMatchSnapshot();
  });
});
