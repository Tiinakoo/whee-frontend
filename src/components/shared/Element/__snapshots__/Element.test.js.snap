// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element renders 1`] = `
<Element>
  <div
    className="element element__minWidth element__padding"
  />
</Element>
`;
